applications/CoSimulationApplication/python_scripts/solver_wrappers/kratos/kratos_base_wrapper.py

```diff
@@ -10,6 +10,24 @@
 import KratosMultiphysics.CoSimulationApplication.co_simulation_tools as cs_tools
 from importlib import import_module
 
+
+class OpenMPThreadManager(object):
+    def __init__(self, num_threads=None):
+        self.num_threads = num_threads
+        if self.num_threads:
+            self.num_threads_orig = KM.OpenMPUtils().GetNumThreads()
+
+    def __enter__(self):
+        if self.num_threads:
+            # print("  >> setting num threads to:", self.num_threads)
+            KM.OpenMPUtils().SetNumThreads(self.num_threads)
```

Comment:
I would change this for `KM.OpenMPUtils().SetNumThreads(min(self.num_threads, self.num_threads_orig))`. Otherwise you may be requesting more resources than you have, own, or the user allow you to use.